websites/Y/YNOproject/presence.ts

```diff
@@ -0,0 +1,67 @@
+const presence = new Presence({
+	clientId: "1028080411772977212",
+});
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/2LD3PQV.png",
+		},
+		titleInfo = document.title.split(" - "),
+		location = document.querySelector("#locationText");
+
+	if (location?.textContent)
+		presenceData.state = `Dreaming at ${location.textContent}`;
+	else presenceData.state = "Dreaming";
+
+	if (titleInfo.length > 1 && titleInfo[1] === "YNOproject" ? true : false) {
+		presenceData.details = `${titleInfo[0]}`;
+		switch (
+			document.querySelector("#modalContainer .modal:not(.hidden) .modalTitle")
+				?.textContent
+		) {
+			case "Settings":
+				presenceData.state = "Changing settings";
+				break;
+			case "Rankings":
+				presenceData.state = "Viewing rankings";
+				break;
+			case "Badges":
+				presenceData.state = "Viewing badges";
+				break;
+			case "Expeditions":
+				presenceData.state = "Viewing expeditions";
+				break;
+			case "UI Theme":
+				presenceData.state = "Changing UI theme";
+				break;
```

Comment:
Is this really needed

Reply:
You mean having a state for the UI theme changer?
I think that if I'm doing a state for all the available buttons in the game page, having one for the UI theme button is fine.